compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -3875,6 +3899,83 @@ private boolean checkReturnValidityInTransaction(AnalyzerData data) {
                 && data.withinTransactionScope;
     }
 
+    private void analyzeInvocationParams(BLangInvocation iExpr, AnalyzerData data) {
+        if (iExpr.symbol == null) {
+            return;
+        }
+
+        BType invocableType = Types.getReferredType(iExpr.symbol.type);
+        BInvokableSymbol invokableSymbol = ((BInvokableSymbol) iExpr.symbol);
+        List<BVarSymbol> reqParamSymbols = invokableSymbol.params;
+        int parameterCountForPositionalArgs = ((BInvokableType) invocableType).getParameterTypes().size();
+
+        int visitedArgCount = 0;
+        for (BLangExpression expr : iExpr.argExprs) {
+            switch (expr.getKind()) {
+                case NAMED_ARGS_EXPR:
+                    reportIfDeprecatedUsage(((BLangNamedArgsExpression) expr).varSymbol, expr, expr.pos);
+                    visitedArgCount++;
+                    break;
+                case REST_ARGS_EXPR:
+                    if (visitedArgCount >= parameterCountForPositionalArgs) {
+                        reportIfDeprecatedUsage(invokableSymbol.restParam, expr, expr.pos);
+                    } else {
```

Comment:
Can reduce indentation by continuing in the if block. Same for the inner if-else block.

Reply:
Same for analyzeRestArgsAgainstReqParams.